src/client/pythonEnvironments/common/pythonBinariesWatcher.ts

```diff
@@ -3,19 +3,23 @@
 
 'use strict';
 
+import * as path from 'path';
+import * as picomatch from 'picomatch';
 import { FileChangeType, watchLocationForPattern } from '../../common/platform/fileSystemWatcher';
 import { getOSType, OSType } from '../../common/utils/platform';
 
 const [executable, binName] = getOSType() === OSType.Windows ? ['python.exe', 'Scripts'] : ['python', 'bin'];
-const patterns = [executable, `*/${executable}`, `*/${binName}/${executable}`];
 
 export function watchLocationForPythonBinaries(
     baseDir: string,
     callback: (type: FileChangeType, absPath: string) => void,
+    executablePattern: string = executable,
 ): void {
+    const patterns = [executablePattern, `*/${executablePattern}`, `*/${binName}/${executablePattern}`];
     for (const pattern of patterns) {
         watchLocationForPattern(baseDir, pattern, (type: FileChangeType, e: string) => {
-            if (!e.endsWith(executable)) {
+            const regex = picomatch.toRegex(executablePattern, { nocase: getOSType() === OSType.Windows });
+            if (!regex.test(path.basename(e))) {

```

Comment:
Using `path.basename()` implies certain restrictions on `executablePattern` that you are not enforcing.  I recommend either failing if `executablePattern.includes(path.separator)` or allowing the broader globs:

```suggestion
            const regex = picomatch.toRegex(path.join('**', executablePattern), { nocase: getOSType() === OSType.Windows });
            if (!regex.test(e)) {
```